File: vaccathon-rest-api/src/main/java/restapi/dto/Patient.java
package restapi.dto;

import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Collection;
import java.util.Date;


@Document
public class Patient extends Person{
    @DBRef
    private Collection<Vaccination> vaccinationList;

    public Patient(Long nationalInsuranceNumber, String firstName, String lastName, Date birthDay, String email, String phone, String city, Collection<Vaccination> vaccinationList) {
        super(nationalInsuranceNumber, firstName, lastName, birthDay, email, phone, city);
        this.vaccinationList = vaccinationList;
    }

    public Collection<Vaccination> getVaccinationList() {
        return vaccinationList;
    }

    public Patient setVaccinationList(Collection<Vaccination> vaccinationList) {
        this.vaccinationList = vaccinationList;
        return this;
    }
}

File: frontend/vaccathon/src/app/models/patient.ts
import {Vaccination} from "./vaccination";
import {Person} from "./person";

export class Patient extends Person{

  constructor(
    public vaccinationList: Vaccination[] = [],
    public  nationalInsuranceNumber: number,
    public  firstName: string,
    public  lastName: string,
    public  birthDay: Date,
    public  email: string,
    public  phone: string,
    public  city: string
  ) {
    super(nationalInsuranceNumber,
      firstName,
      lastName,
      birthDay,
      email,
      phone,
      city);
  }


}
